File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = {} #value -> count

        for val in nums :
            counter[val] = counter.get(val,0) + 1
        
        print(counter)
        arr = [[]] * len(nums)

        for val in counter :
            arr[counter[val]].append(val)
            print(arr)

        return  arr



            

File: Data Structures & Algorithms/search-insert-position/submission-2.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int l=0, r=nums.length-1;
        
        while(l<r){
            int middle = (l+r)/2;
            
            if(nums[middle] > target){
                r = middle-1;
            }else if (nums[middle] < target){
                l = middle+1; 
            }else{
                return middle;
            }
        }
        
        System.out.println("l & r =" + l + r);
        
        return l;
    }
}